source/jobs.c

```diff
@@ -430,19 +431,39 @@ static JobsStatus_t matchIdApi( char * topic,
     if( ( isNextJobId( jobId, jobIdLength ) == true ) ||
         ( isValidJobId( jobId, jobIdLength ) == true ) )
     {
-        JobsTopic_t api;
+        int32_t api;
 
         /* The api variable is bounded within contiguous values of the enum type. */
-        /* coverity[misra_c_2012_rule_10_1_violation] */
-        for( api = JobsDescribeSuccess; api < JobsMaxTopic; api++ )
+        for( api = ( int32_t ) JobsDescribeSuccess; api < ( int32_t ) JobsMaxTopic; api++ )
         {
             ret = strnnEq( p, length, apiTopic[ api ], apiTopicLength[ api ] );
 
             if( ret == JobsSuccess )
             {
-                *outApi = api;
                 *outJobId = jobId;
                 *outJobIdLength = jobIdLength;
+
+                /* It is a MISRA violation to cast an int32_t to an enum, so switch statement the return */
+                switch( api )
```

Comment:
using an if else is more explicit and less confusing for whats going on here. the whole purpose of this switch statement is just to cast the final api found and return it, right? so why not just if else check all possibilities. a lot less confusing than for looping through them because they are contiguous and then using a switch statement to cast it. using if else you can get rid of the api variable altogether.

Reply:
Swapped to if/else if statements. 
One part because of discussion about the loops, another part because the use of the switch statement moved the matchIdApi() to a 15 on complexity checking, when it is currently at a 5.